plugins/gradle/gradle-plugin/src/main/java/org/wildfly/swarm/plugin/gradle/StartTask.java

```diff
@@ -0,0 +1,354 @@
+import org.gradle.api.tasks.bundling.War;
+import org.gradle.initialization.BuildCancellationToken;
+import org.wildfly.swarm.bootstrap.util.BootstrapProperties;
+import org.wildfly.swarm.fractions.FractionDescriptor;
+import org.wildfly.swarm.fractions.FractionList;
+import org.wildfly.swarm.fractions.FractionUsageAnalyzer;
+import org.wildfly.swarm.tools.ArtifactResolvingHelper;
+import org.wildfly.swarm.tools.ArtifactSpec;
+import org.wildfly.swarm.tools.BuildTool;
+import org.wildfly.swarm.tools.DependencyManager;
+import org.wildfly.swarm.tools.exec.SwarmExecutor;
+import org.wildfly.swarm.tools.exec.SwarmProcess;
+
+public class StartTask extends DefaultTask {
+
+    private static final String EXCLUDE_PREFIX = "!";
+    private static final String INDENTATION_DELIMITER = "\n  ";
+
+    private ThorntailExtension extension;
+    private boolean waitForProcess = false;
+
+    public StartTask() {
+        extension = getProject().getExtensions().getByType(ThorntailExtension.class);
+    }
+
+    void waitForProcess() {
+        waitForProcess = true;
+    }
+
+    @TaskAction
+    public void startApplication() {
+        final SwarmExecutor executor;
+        if (extension.isUseUberJar()) {
+            executor = uberJarExecutor();
+        } else if (!getProject().getTasks().withType(War.class).isEmpty()) {
+            executor = warExecutor();
+        } else if (!getProject().getTasks().withType(Jar.class).isEmpty()) {
+            executor = jarExecutor();
+        } else {
+            throw new GradleException("Unable to determine type of the project, neither war or jar tasks were found.");
+        }
+
+        executor.withJVMArguments(extension.getJvmArguments());
+        executor.withArguments(extension.getArguments());
+
+        final SwarmProcess process;
+        try {
+
+            File tmp;
+            try {
+                tmp = Files.createTempFile("thorntail-process-file", null).toFile();
+                getLogger().info("Process file: " + tmp.toPath().toString());
+            } catch (IOException e) {
+                throw new GradleException("Error while creating process file");
+            }
+
+            process = executor.withDebug(extension.getDebugPort())
+                    .withProcessFile(tmp)
+                    .withProperties(extension.getProperties())
+                    .withProperties(ThorntailUtils.getPropertiesFromFile(extension.getPropertiesFile()))
+                    .withStdoutFile(extension.getStdoutFile() != null ? getProject().file(extension.getStdoutFile()).toPath() : null)
+                    .withStderrFile(extension.getStderrFile() != null ? getProject().file(extension.getStderrFile()).toPath() : null)
+                    .withEnvironment(extension.getEnvironment())
+                    .withEnvironment(ThorntailUtils.getPropertiesFromFile(extension.getEnvironmentFile()))
+                    .withWorkingDirectory(getProject().getProjectDir().toPath())
```

Comment:
Yeah, basically my intention was to keep it safe by following the maven plugin behavior as closely as possible, and maven plugin appears to be using project dir:
https://github.com/thorntail/thorntail/blob/master/plugins/maven/src/main/java/org/wildfly/swarm/plugin/maven/StartMojo.java#L133

The build dir might be better in case that the process decides to create some files, so that it doesn't pollute the project. One think I could test whether this affects location of the log files in case when std or err output file is set to relative path. Will do that tomorrow.

Reply:
So setting stdoutFile in thorntail extension actually always sets an absolute file, so working directory has no effect here. I would keep WD = project dir to stay consistent with the maven plugin.

```
thorntail {
  stdoutFile = file('stdout.log') // or new File('stdout.log')
  println stdoutFile.absolutePath // prints <projectDir>/stdout.log
}
```

Reply:
Keeping WS = project dir to stay consistent with maven plugin.